### ISSUE TITLE
[bug] leading and trailing comments incorrect for bin

### ISSUE THREAD

**Issue by evilebottnawi** [glayzzle/php-parser#189]
Input:
```php
<?php

$var = 'string1'
    // Comment 1
    . 'string2' // Comment 2
    // Comment 3
    . 'string3';
```

Here `Comment 1` belong to `bin` node as `leadingComment`, but should belong to `string2` (`string` node). Also looks some node doesn't have `trailingComments` and `leadingComment` comments.

Looks how babylon do this for binary nodes https://astexplorer.net/#/gist/2674d53a98c68e273b96adaa99047e61/5c6bbdbfad0d9996203e56dcfae8659ec593fc57

Thanks!

**Comment by ichiriac** [comment]
the fix is released under `php-parser@3.0.0-prerelease.5`

**Comment by ichiriac** [comment]
but it's not exactly the same implementation as babylon as the `trailingComments` property is not handled. I'll open a new issue around this as a feature request
